src/net_processing.h

```diff
@@ -6,19 +6,13 @@
 #ifndef BITCOIN_NET_PROCESSING_H
 #define BITCOIN_NET_PROCESSING_H
 
-#include <consensus/params.h>
 #include <net.h>
 #include <sync.h>
-#include <txrequest.h>
 #include <validationinterface.h>
 
-class BlockTransactionsRequest;
-class BlockValidationState;
-class CBlockHeader;
 class CChainParams;
 class CTxMemPool;
 class ChainstateManager;
-class TxValidationState;
 
 extern RecursiveMutex cs_main;
 extern RecursiveMutex g_cs_orphans;
```

Comment:
This only needs to be exposed because of the weird lock ordering requirements documented here: https://github.com/bitcoin/bitcoin/blob/e75f91eae3936269b40b4bfdfe540d5526270936/src/init.cpp#L202-L208. Once tx data is moved into `Peer` and `RelayTransactions()` no longer needs to call `ForEachNode()` and therefore take `cs_vNodes` while holding `g_cs_orphans`, this mutex can be private in net_processing.

Reply:
I don't think that's relevant for this PR though?

Reply:
Correct, this was just an observation about why is required here now.